src/Components/Artwork/GridItem.tsx

```diff
@@ -30,10 +34,14 @@ class ArtworkGridItemContainer extends React.Component<Props, null> {
   }
 
   render() {
-    const { style, className, artwork, useRelay } = this.props
+    const { style, className, artwork, useRelay, currentUser } = this.props
     const SaveButtonBlock = useRelay ? RelaySaveButton : SaveButton
     const MetadataBlock = useRelay ? RelayMetadata : Metadata
 
+    let currentUserSpread = {}
```

Comment:
This is where we don't want to overwrite an injected `currentUser`, in a legacy use of the grid/save button, with a new one we are passing down.

Listening to `AppState` here isn't possible without a breaking change (the way this is currently being used in Force doesn't go thru the current flow that sets that up properly).